docs/3.0.0-beta.x/guides/queries.md

```diff
@@ -0,0 +1,406 @@
+            _.endsWith(association.nature, 'One') ||
+            association.nature === 'oneToMany'
+              ? { [association.via]: data._id }
+              : { [association.via]: { $in: [data._id] } };
+          const update =
+            _.endsWith(association.nature, 'One') ||
+            association.nature === 'oneToMany'
+              ? { [association.via]: null }
+              : { $pull: { [association.via]: data._id } };
+
+          // Retrieve model.
+          const model = association.plugin
+            ? strapi.plugins[association.plugin].models[
+                association.model || association.collection
+              ]
+            : strapi.models[association.model || association.collection];
+
+          return model.update(search, update, { multi: true });
+        })
+      );
+
+      return data;
+    },
+
+    search(params, populate) {
+      // Convert `params` object to filters compatible with Mongo.
+      const filters = strapi.utils.models.convertParams(model.globalId, params);
+
+      const $or = buildSearchOr(model, params._q);
+
+      return model
+        .find({ $or })
+        .sort(filters.sort)
+        .skip(filters.start)
+        .limit(filters.limit)
+        .populate(populate || defaultPopulate);
+    },
+
+    countSearch(params) {
+      const $or = buildSearchOr(model, params._q);
+      return model.find({ $or }).countDocuments();
+    },
+  };
+};
+
+const buildSearchOr = (model, query) => {
+  return Object.keys(model.attributes).reduce((acc, curr) => {
+    switch (model.attributes[curr].type) {
+      case 'integer':
+      case 'float':
+      case 'decimal':
+        if (!_.isNaN(_.toNumber(query))) {
+          return acc.concat({ [curr]: query });
+        }
+
+        return acc;
+      case 'string':
+      case 'text':
+      case 'password':
+        return acc.concat({ [curr]: { $regex: query, $options: 'i' } });
+      case 'boolean':
+        if (query === 'true' || query === 'false') {
+          return acc.concat({ [curr]: query === 'true' });
+        }
+
+        return acc;
+      default:
+        return acc;
+    }
+  }, []);
+};
+```
+
+## Understanding queries
+
+`strapi.query` will generate a queries object by passing a model to the factory function matching the model's ORM.
+
+In this example the User model from the Users and Permissions plugin is used.
+By default the model is passed to `strapi/lib/core-api/queries/bookshelf.js` or `strapi/lib/core-api/queries/mongoose.js` depending on your connection configuration.
+
+```js
+const queries = strapi.query('users', 'users-permissions');
+
+// makes the bookshelf or mongoose queries available with a specific model binded to them
+queries.find();
+```
+
+### Usage in plugins
+
+To make plugins ORM agnostic, we create a queries function for every plugin that will either load the queries from the plugin's `config/queries` folder if it exists or use the default queries from the `core-api/queries` folder.
```

Comment:
To make plugins ORM agnostic, a queries function is created for every plugin to either load the queries from the plugin's `config/queries` folder (if it exists) or to use the default queries from the `core-api/queries` folder.